vscode/src/autoedits/renderer/inline-manager.ts

```diff
@@ -74,14 +74,25 @@ export class AutoEditsInlineRendererManager
     protected async onDidChangeTextEditorSelection(
         event: vscode.TextEditorSelectionChangeEvent
     ): Promise<void> {
-        if (this.hasInlineCompletionItems() && isFileURI(event.textEditor.document.uri)) {
-            // We are showing a completion, possibly alongisde decorations. The dismissal of the completion will
+        if (
+            this.hasInlineCompletionItems() &&
+            // If we do not have inline decorations, the dismissal of the completion will
+            // automatically be handled by VS Code. Without this check, we will falsely
+            // reject an inline completion, because the `onDidChangeTextEditorSelection` callback
+            // can be fired before the inline-completion acceptance callback.
+            //
+            // I was not able to reproduce it in the E2E tests. For some reason the order of callbacks
+            // is different.
+            this.hasInlineDecorations() &&
+            areSameUriDocs(event.textEditor.document, this.activeRequest?.document)
+        ) {
```

Comment:
```suggestion
        if (
            this.activeRequest?.renderOutput.type === 'completion-with-decorations' &&
            areSameUriDocs(event.textEditor.document, this.activeRequest?.document)
        ) {
```

This is the same I think, we just need to handle special cases where we are also showing a completion and we can't use our preferred hiding logic. You can delete `hasInlineCompletionItems`, I think this was an oversight on my part. The comment I added doesn't match up with the behaviour (doing this for pure completions too)